src/common/components/header/index.js

```diff
@@ -7,7 +7,7 @@ import styles from './header.css';
 
 const wordmark = 'https://assets.ubuntu.com/v1/d45097a4-snapcraft.io-logotype.svg';
 
-class Header extends Component {
+export class Header extends Component {
```

Comment:
So I think we're all aiming for the same things, a nice, simple developer environment, that causes us as little bother as possible and allows as much energy to be put on design as possible.

In terms of design, HoC are a form of composition, and for me, breaking components up at this stage is over thinking the problem, but I may not fully understand the problem (I'm somewhat trying to understand it from the solutions people are providing).

Limiting to single exports I can't see the real benefit of at this time, so my vote is to export the component as a named export, the wrapped component as the default and test the named export. Unit testing the wrapped component is really just testing Redux's connect, and it already has unit tests.

We also have code style guides but they're well out of date, as we've been in a constant crunch for over 6 months now. These should be merged into the webteam repo https://github.com/canonical-ols/style-guides